File: SearchInRotatedSortedArray/I/SearchInRotatedSortedArray.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int flag = nums.size(), left, right, mid;
        for(int i = 0; i < nums.size() - 1; i++) {
        	if(nums[i] > nums[i + 1]) {
        		flag = i + 1;
        		break;
        	}
        }
        if(target >= nums[0]) {
        	left = 0;
        	right = flag;
        }
        else {
        	left = flag;
        	right = nums.size();
        }
        while(left < right) {
        	mid = left + (right - left) / 2;
        	if(target == nums[mid])
        		return mid;
        	else if(target < nums[mid])
        		right = mid;
        	else
        		left = mid + 1;
        }
        return -1;
    }
};

File: SearchInRotatedSortedArray/II/SearchInRotatedSortedArray.py
class Solution(object):
    def search(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: bool
        """
        flag = len(nums)
        for i in range(len(nums) - 1):
            if nums[i] > nums[i + 1]:
                flag = i + 1
        if target < nums[0]:
            left = flag
            right = len(nums) - 1
        else:
            left = 0
            right = flag - 1
        while left <= right:
            mid = left + (right - left) / 2
            if nums[mid] == target:
                return True
            elif nums[mid] > target:
                right = mid - 1
            else:
                left = mid + 1
        return False